src/mockServiceWorker.js

```diff
@@ -97,7 +97,7 @@ self.addEventListener('fetch', async function (event) {
         // Bypass mocking for navigation requests
         request.mode === 'navigate'
       ) {
-        return resolve(getOriginalResponse())
+        return
```

Comment:
Is it ok that the promise is never resolved in that case? Why we can not do an early return before the `event.respondWith`?

Reply:
Early return before `event.respondWith` throws an exception that request cannot be handled because the event is closed. 

Good point about pending Promise. I'm not sure if it's an issue, though. We shouldn't resolve it, that will cause a fake response without any data.

Reply:
I'm not sure that it will work. Intercepting the `fetch` I think that we should provide a way to resolve the request, fetching from original one or from cache, or mocking as our case.

Reply:
The fact that it performs the request may imply that it actually works as intended. Had the promise given to `respondWith` been pending forever, we'd receive a forever pending request in the network.

That being said, there are some failing tests on CI with a navigation timeout. This may be caused by multiple factors, including forever pending requests. I haven't encountered those while manually testing.

Reply:
You are right: not resolving a Promise here causes an infinitely pending request. I haven't spotted this due to some race conditions between the worker registration and the request dispatch. Now I can see that.